addons/website_profile/controllers/main.py

```diff
@@ -37,13 +37,17 @@ def _check_avatar_access(self, user_id, **post):
 
     def _check_user_profile_access(self, user_id):
         user_sudo = request.env['res.users'].sudo().browse(user_id)
+        result = {'denial': False, 'not_found': False, 'user': user_sudo}
         # User can access - no matter what - his own profile
         if user_sudo.id == request.env.user.id:
-            return user_sudo
-        if user_sudo.karma == 0 or not user_sudo.website_published or \
-            (user_sudo.id != request.session.uid and request.env.user.karma < request.website.karma_profile_min):
-            return False
-        return user_sudo
+            return result
+        if not user_sudo.exists():
+            result['not_found'] = True
+        elif user_sudo.karma == 0 or not user_sudo.website_published:
```

Comment:
What is this 0? A hardcoded value for users to have their profile public?
Should we have a separate answer in this case?

Reply:
but we check for zero value, and not `<= 0`, so I think @flch-odoo assumption is more correct

@dbeguin The commit is a bit old (4 year ago), but do you remember why we check for `user_sudo.karma == 0` ? (and not `user_sudo.karma <= 0` or just ignoring the karma of the user we are trying to access)

Reply:
@std-odoo @dbeguin @flch-odoo I think I have a match with `user_sudo.karma == 0`.
When users sign up, they start with a karma of `0` and are invited to have their profile validated (email verification). On performing profile validation users get `3` points of karma. see odoo/odoo#31321
So `user_sudo.karma == 0` would be to prevent access to unverified  profile. If so, do you think we should have a specific denial reason for that case?

Reply:
I don't think so. Private fits for both cases. However, is it possible to check something else than karma for this?
Otherwise you can get exactly -100 when you were at 100 and then become muted while if you had 99 or 101 you wouldn't.

Reply:
I agree with FLorian, it's a bit weird to use the karma for that (if it was used for that reason)

Reply:
we will keep it like this for now, but I think the validation email flow should be cleaned (see message in the task chatter)